modin/logging/logger_decorator.py

```diff
@@ -143,3 +144,66 @@ def run_and_log(*args: Tuple, **kwargs: Dict) -> Any:
         return disable_logging(run_and_log)
 
     return decorator
+
+
+def enable_remote_logging(
+    func: Callable,
+    modin_layer: str = "RAY-REMOTE",
+    log_level: str = "info",
+) -> Callable:
+    """
+    Log Decorator used on Modin remote functions.
+
+    Parameters
+    ----------
+    func : Callable
+        The function to decorate.
+    modin_layer : str, default: "RAY-REMOTE"
+        The modin_layer to decorate the remote function with.
+    log_level : str, default: "info"
+        The log level (INFO, DEBUG, WARNING, etc.).
+
+    Returns
+    -------
+    func
+        A decorator function.
+    """
+    log_level = log_level.lower()
+    assert hasattr(Logger, log_level.lower()), f"Invalid log level: {log_level}"
+
+    @wraps(func)
+    def run_and_log(*args: tuple, **kwargs: Dict[Any, Any]) -> None:
+        """
+        Compute function with logging if Modin logging is enabled.
+
+        Parameters
+        ----------
+        *args : tuple
+            The function arguments.
+        **kwargs : dict
+            The function keyword arguments.
+
+        Returns
+        -------
+        Any
+        """
+        # Don't log remote calls if LogMode is disable or api_only.
+        if LogMode.get() != "enable":
+            return func(*args, **kwargs)
+
+        logger = get_worker_logger(JOB_ID)
+        start_line = f"START::{modin_layer.upper()}::{func.__name__}"
+        stop_line = f"STOP::{modin_layer.upper()}::{func.__name__}"
```

Comment:
these two should be computed outside of `run_and_log` (see `enable_logging` implementation), they never change, we shouldn't be wasting time computing same things over and over again

Reply:
this feels super weird, got a stacktrace?..

Reply:
https://gist.github.com/jeffreykennethli/4cd7606e68718000b9467e9f47b6c9a0. also see https://github.com/jeffreykennethli/modin/blob/log_engine_2/modin/logging/logger_decorator.py#L149-L215